src/scenes/Esittelijat/modules/asiatUtils.js

```diff
@@ -39,19 +39,20 @@ const generateAsiatTableHeaderStructure = t => {
   };
 };
 
-const getMaakuntaNimiFromHakemus = hakemus => {
-  const maakuntaObject = R.find(R.propEq("kieli", "FI"))(
+const getMaakuntaNimiFromHakemus = (hakemus, locale) => {
+  const maakuntaObject = R.find(R.propEq("kieli", locale.toUpperCase()))(
     R.path(["jarjestaja", "maakuntaKoodi", "metadata"], hakemus) || []
   );
   return maakuntaObject ? maakuntaObject.nimi : "";
 };
 
-const getJarjestajaNimiFromHakemus = hakemus => {
-  return R.path(["jarjestaja", "nimi", "fi"], hakemus) || "";
+const getJarjestajaNimiFromHakemus = (hakemus, locale) => {
+  const nimi = R.path(["jarjestaja", "nimi"], hakemus);
+  return R.prop(locale, nimi) || R.last(R.values(nimi));
 };
 
 // Generates common row data for all Asiat-tables
-export const generateAsiaTableRows = (row, i, t) => {
+export const generateAsiaTableRows = (row, i, {formatMessage, locale}) => {
```

Comment:
i can be removed. It's not used.

Reply:
for a brief moment this comment was quite confusing until i realised _i_ references to a parameter  :)

Reply:
This made my day. :D